### ISSUE TITLE
Allow `terraform state mv` item to a new list

### ISSUE THREAD

**Issue by yujunz** [hashicorp/terraform#23497]
### Current Terraform Version

```
Terraform v0.12.10
```

### Use-cases

I want to rename a **list** of resources and it seems not supported if the target list does not exist.

```sh
❯ terraform@0.12 state mv aws_instance.{k8s,cpu_node}\[0\]
Move "aws_instance.k8s[0]" to "aws_instance.cpu_node[0]"

Error: Invalid target address

Cannot move to aws_instance.cpu_node[0]: aws_instance.cpu_node does not exist
in the current state.
```

### Attempted Solutions

I didn't find a way to workaround it.

### Proposal

Support creating the list when moving resource in state to a new list.

### References

N/A

**Comment by danieldreier** [comment]
@yujunz thank you for proposing this improvement. We're going to look at this use case on our end and triage this. It would be really helpful if you would provide more context on your larger workflow so that we can understand why you want to manipulate state like this.

**Comment by yujunz** [comment]
tl;dr rename a list of resources during refactoring

1. A list of `aws_instance` called `k8s` is created as general purpose nodes
2.Later on, we want to add a GPU node to the cluster. So another list of `aws_instance` is created, named `gpu_node`
3. A GPU node is literally a k8s node. So I want to change the name `k8s` to `cpu_node` to avoid confusion.

**Comment by jbardin** [comment]
Closed by #23582
